src/utils/transaction/insertTransaction.ts

```diff
@@ -58,98 +152,70 @@ export const insertTransaction = async (
     target: getChecksumAddress(insertedTransaction.target),
     sender: getChecksumAddress(insertedTransaction.sender),
     value: insertedTransaction.value ?? 0n,
+    walletType: "local", // we override this later
   };
 
+  // First attempt: try to find wallet by 'from' address
   let walletDetails: ParsedWalletDetails | undefined;
+  let walletFoundByFrom = false;
 
   try {
     walletDetails = await getWalletDetails({
-      address: queuedTransaction.from,
+      walletAddress: queuedTransaction.from,
     });
+    walletFoundByFrom = true;
+  } catch {}
 
-    // when using the v5 SDK with smart backend wallets, the following values are not set correctly:
-    // isUserOp is set to false
-    // account address is blank or the user provided value (this should be the SBW account address)
-    // from is set to the SBW account address (this should be the SBW signer address)
-    // these values need to be corrected so the worker can process the transaction
+  // Case 1 & 2: Wallet found by 'from'
+  if (walletFoundByFrom && walletDetails) {
```

Comment:
Shouldn't these be the same? If walletDetails is truthy, walletFoundByFrom must be true here.